executor/aggregate.go

```diff
@@ -827,33 +830,54 @@ func (e *StreamAggExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	return nil
 }
 
-func (e *StreamAggExec) consumeOneGroup(ctx context.Context, chk *chunk.Chunk) error {
-	for !e.executed {
-		if err := e.fetchChildIfNecessary(ctx, chk); err != nil {
+func (e *StreamAggExec) consumeOneGroup(ctx context.Context, chk *chunk.Chunk) (err error) {
+	if e.groupChecker.isExhausted() {
+		if err = e.fetchChild(ctx, chk); err != nil {
 			return err
 		}
-		for ; e.inputRow != e.inputIter.End(); e.inputRow = e.inputIter.Next() {
-			meetNewGroup, err := e.groupChecker.meetNewGroup(e.inputRow)
+		if !e.executed && !chk.IsFull() {
+			_, err := e.groupChecker.splitIntoGroups(e.childResult)
 			if err != nil {
 				return err
 			}
-			if meetNewGroup {
-				err := e.consumeGroupRows()
-				if err != nil {
-					return err
-				}
-				err = e.appendResult2Chunk(chk)
-				if err != nil {
-					return err
-				}
-			}
-			e.groupRows = append(e.groupRows, e.inputRow)
-			if meetNewGroup {
-				e.inputRow = e.inputIter.Next()
-				return nil
+		} else {
+			return nil
+		}
+	}
+	begin, end := e.groupChecker.getNextGroup()
+	e.groupRows = e.groupRows[:0]
+	for i := begin; i < end; i++ {
+		e.groupRows = append(e.groupRows, e.childResult.GetRow(i))
+	}
+
+	for end == e.childResult.NumRows() {
+		end = -1
+		if err = e.fetchChild(ctx, chk); err != nil || e.executed {
```

Comment:
e.childResult will be reset in this function, will it affect the e.groupRows built in line 850?

Reply:
No. In function ```fetchChild```, it will solve the ```e.childResult``` and update the result. See [here](https://github.com/pingcap/tidb/blob/445e30204a463b8a370e950829c34d6fd16cd4d6/executor/aggregate.go#L881) and [here](https://github.com/pingcap/tidb/blob/445e30204a463b8a370e950829c34d6fd16cd4d6/executor/aggregate.go#L860:25) for more details.